File: vehicle-recommendation-service/models/travel.py
from typing import Optional
from pydantic import BaseModel, Field
from bson import ObjectId


class Travel(BaseModel):
    id: Optional[str] = Field(
        default_factory=lambda: str(ObjectId()), alias="_id")
    from_: str
    to: str
    is_active: bool = True
    departure_at: str
    route: str
    driver_no: int
    vehicle_id: str
    seat_average_price: int
    created_at: Optional[str] = None
    updated_at: Optional[str] = None

File: user-client/schema/Travel.ts
import z from "zod";

export const travelSchema = z.object({
  _id: z.string(),
  from: z.string(),
  to: z.string(),
  is_active: z.boolean().default(true),
  seat_average_price: z.number(),
  departure_at: z.string(),
  route: z.string().nullable(),
  driver_no: z.number(),
  vehicle_id: z.string(),
});

export type Travel = z.infer<typeof travelSchema>;
